src/provider/model/AppWindow.ts

```diff
@@ -62,4 +65,18 @@ export class AppWindow {
     public focus(): Promise<void> {
         return this._window.setAsForeground();
     }
+
+    public isReadyToReceiveIntent(intent: IntentType): Promise<void> {
```

Comment:
Is there a better name for this? Suggests returns a boolean

Reply:
True, I changed it to `ensureReadyToReceiveIntent`